### DISCUSSION TITLE
How Can I Set a Fixed Y-Axis Range in Vico?

### DISCUSSION THREAD

**Discussion by MonteCreasor** [patrykandpatrick/vico#944] [Questions] [answered]
### Question

Hi, I'm using Vico to plot data that updates frequently, around three times per second. The x-axis values remain consistent, so there’s no issue there. However, my y-axis values vary significantly, sometimes reaching a max of 30–40 and other times up to 1200. This causes the y-axis scale to adjust dynamically with each update, resulting in a distracting "jumping" effect on the chart.

Is there a way to lock or set a fixed range for the y-axis (e.g., 0 to 1200) to stay consistent regardless of the incoming data range? Any guidance on achieving this would be greatly appreciated.

Thanks!

### Vico version(s)

2.0.0-beta.2

**Comment by MonteCreasor** [comment, marked as answer]
I figured it out just now ... CartesianLayerRangeProvider

**Reply by patrickmichalik** [reply]
Thanks for the follow-up!

**Comment by MonteCreasor** [comment]
No problem.

The app has a steep learning curve if you want to do more than just draw a
simple graph, but after diving into this issue and your source code, it
became obvious to me that your app is a great example of clever engineering
using really great composition principles. I'm 65 this year and been doing
software since university and I know great code when I see it. Good job
really!

**Reply by patrickmichalik** [reply]
Thanks for the kind words! Being designed for flexibility and extensibility, Vico does come with a slightly steeper learning curve. We recommend referring to [the guide](https://www.patrykandpatrick.com/vico/guide/stable), which provides a high-level overview and covers topics like this one. Since this discussion was opened, we’ve also introduced more sample charts demonstrating `CartesianLayerRangeProvider`, including its `fixed` factory function.
